File: src/main/kotlin/a_basic_control_structures/exercise_6.kt
package a_basic_control_structures

// Write a program that reads an integer and displays its predecessor and successor.

fun main() {
    print("Write a number: ")
    val number = readln().toInt()

    val predecessor = number - 1
    val successor = number + 1

    println("The predecessor of $number is $predecessor; the successor of $number is $successor.")
}

File: Dart/a_basic_control_structure/exercise_6.dart
// Write a program that reads an integer and displays its predecessor and successor.

import 'dart:io';

void main() {
  tryAgain();
}

void predecessorAndSucessor(int number) {
  int predecessor = number - 1;
  int successor = number + 1;

  print(
    'the predecessor of $number is $predecessor; the successor of $number is $successor.',
  );
}

void tryAgain() {
  String answer;

  do {
    stdout.write('write a number: ');
    try {
      int number = int.parse(stdin.readLineSync()!);
      predecessorAndSucessor(number);
    } catch (exception) {
      print('Error: Please enter a valid integer');
    }

    stdout.write('do you want to try again? ');
    answer = stdin.readLineSync()!;
  } while (answer.toLowerCase() == 'yes');
}
